File: MontessoriSystem/montessoriSystem.core.Application/ViewModels/Attendance/AttendanceSummaryViewModel.cs
using MontessoriSystem.Core.Application.ViewModels.Common;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MontessoriSystem.Core.Application.ViewModels.Attendance
{
    public class AttendanceSummaryViewModel:AuditableBaseModel
    {
        public List<AttendanceViewModel> Attendances { get; set; }
        public int PresentCount { get; set; }
        public int DelayCount { get; set; }
        public int AbsentCount { get; set; }
        public int ExcuseCount { get; set; }
        public int TotalCount { get; set; }
    }
}

File: Smart-Student-React/src/interfaces/Attendace/AttendanceSummaryViewModel.ts
import { AuditableBaseModel } from "../AuditableBaseModel";



export interface AttendanceSummaryViewModel extends AuditableBaseModel {
    attendances: AttendanceViewModel[];
    presentCount: number;
    delayCount: number;
    absentCount: number;
    excuseCount: number;
    totalCount: number;
}
export class AttendanceSummaryViewModelConvert {
    public static toAttendanceSummaryViewModel(json: string | object): AttendanceSummaryViewModel {
        if (typeof json === "string") {
            return JSON.parse(json);
        }
        return json as AttendanceSummaryViewModel;
    }
}



export interface AttendanceViewModel extends AuditableBaseModel {
    date: Date;
    observation?: string;
    isPresent?: boolean;
    isDelay?: boolean;
    isAbsent?: boolean;
    isExcuse?: boolean;

    // Custom
    fullNameStudent?: string;
    idRoom: number;
    idStudent: number;
    idSuject?: number;
    numberList: number;
}
